engine/src/inputs/controller.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023-present Genesis Engine contributors (see LICENSE.txt)
+
+#include "inputs/controller.hpp"
+
+#include <vector>
+#include <map>
+#include <string>
+#include <iostream>
+
+namespace gen
+{
+	namespace controllerManager
+	{
+		namespace
+		{
+			std::vector<Controller> g_connectedControllers; //Max 16 (because of GLFW)
```

Comment:
Why not use an array of optionals? That way the index will match GLFW's ID, whereas this way a controller that randomly disconnects will shift the IDs of all that follow.

Reply:
kinda does the same here because when a controller disconnects I don't delete it from the vector, I just set it as inactive

Reply:
Gotcha, makes sense!